vladiate/inputs.py

```diff
@@ -68,12 +76,32 @@ def __init__(self, path=None, bucket=None, key=None):
             )
 
     def open(self):
-        s3 = self.boto.connect_s3()
+        # aws_access_key_id, aws_secret_access_key
+        if self.aws_config:
+            key = self.aws_config["aws_access_key_id"]
+            secret = self.aws_config["aws_secret_access_key"]
+            s3 = self.boto.connect_s3(
+                aws_access_key_id=key, aws_secret_access_key=secret
+            )
+        else:
+            s3 = self.boto.connect_s3()
```

Comment:
Since `self.aws_config` is a dictionary, and the default values for these kwargs is `None` this can be simplified:

```python
def open(self):
    s3 = self.boto.connect_s3(
        aws_access_key_id=self.aws_config.get("aws_access_key_id"),
        aws_secret_access_key=self.aws_config.get("aws_secret_access_key"),
    )
```

Reply:
Done.